pkg/server/skydb/access.go

```diff
@@ -230,51 +231,58 @@ type FieldACLIterator struct {
 	recordType  string
 	recordField string
 
-	nextRecordTypes []string
-	nextEntries     []FieldACLEntry
-	eof             bool
+	nextEntries []FieldACLEntry
 }
 
 // NewFieldACLIterator creates a new iterator.
 func NewFieldACLIterator(acl FieldACL, recordType, recordField string) *FieldACLIterator {
+	// Categorize three types of entry list
```

Comment:
I understand that you want to categorise all the ACE in one of the three categories, and then return the category based on `recordType` and `recordField`.

Since the FieldACL is sorted already, the iterator doesn’t need to process all the ACEs first in order to find the required matches. If the iterator found a matched ACE, it can keep returning the matches ACE until the next ACE doesn’t match again.

i.e.

1. for each ACE, check if record type and record field matches the record
2. if matches, return the ACE, and remember the record type and record field pair
3. keep returning the ACE that has the same record type and record field pair
4. when the next ACE doesn't have the same record type and record field pair, return nil and stop iterating

would be great if you could change it back